### PR TITLE
Build list of URLs simpler

### PR DESCRIPTION
**Pull request by deldesir**
Replaces list comprehension. Makes code simpler.

### CODE REVIEW

File: cps/tasks/metadata_extract.py [hunk 1]
```diff
@@ -67,9 +67,12 @@ def _fetch_requested_urls(self, conn):
...
             rows = conn.execute(query).fetchall()
-            requested_urls = {row[0]: {"duration": row[1]} if row[1] is not None and row[1] > 0 else self.unavailable.append(row[0]) for row in rows}
-            for url in self.unavailable:
-                requested_urls.pop(url)
+            requested_urls = {}
+            for path, duration in rows:
+                if duration is not None and duration > 0:
+                  requested_urls[path] = {"duration": duration};
+               else:
+                  self.unavailable.append(path)
```
Comment (holta) [inline cps/tasks/metadata_extract.py:75]:
```suggestion
                if duration is not None and duration > 0:
                    requested_urls[path] = {"duration": duration};
                else:
                    self.unavailable.append(path)
```

Comment (deldesir) [reply cps/tasks/metadata_extract.py]: My bad, I committed using my android. Was done badly.

**Comment by holta** [comment]
Is this PR tested?

(Please validate/confirm similar to original PR... !)

- PR #179

**Comment by deldesir** [comment]
Tested on Ubuntu 24.04 (LRN2)

image

File: cps/tasks/metadata_extract.py [hunk 2]
```diff
@@ -67,9 +67,12 @@ def _fetch_requested_urls(self, conn):
                      if "error" in self.columns
                      else "SELECT path, duration FROM media WHERE path LIKE 'http%'")
             rows = conn.execute(query).fetchall()
-            requested_urls = {row[0]: {"duration": row[1]} if row[1] is not None and row[1] > 0 else self.unavailable.append(row[0]) for row in rows}
-            for url in self.unavailable:
-                requested_urls.pop(url)
+            requested_urls = {}
+            for path, duration in rows:
+                if duration is not None and duration > 0:
+                    requested_urls[path] = {"duration": duration};
```
Comment (holta) [inline cps/tasks/metadata_extract.py:73]: > requested_urls[path] = {"duration": duration};

@deldesir @codewiz is the trailing semicolon (above) truly necessary?

Comment (codewiz) [reply cps/tasks/metadata_extract.py:73]: No, that was my mistake... I guess for doing too much C++

Comment (deldesir) [reply cps/tasks/metadata_extract.py:73]: No. I just removed it.

Comment (holta) [reply cps/tasks/metadata_extract.py:73]: Thanks:

https://github.com/iiab/calibre-web/commit/b0adaa0b26e984d3048a4183cf280dcb4a99d3e9

**Pull request merged**